source/dlangbot/database.d

```diff
@@ -0,0 +1,54 @@
+/**
+Implements a database to keep track of the points scored
+by authors that fix bugs. The table has the following
+columns:
+
+Github account (Primary key)
+Total points
+*/
+module dlangbot.database;
+
+import ae.sys.database : Database;
+
+private Database _db;
+enum table_schema = q"SQL
+CREATE TABLE [LEADERBOARD] (
+[AUTHOR]  TEXT PRIMARY KEY NOT NULL,
+[POINTS]  INTEGER
+);
+SQL";
+
+/**
+Checks whether a database file already exists.
+If that is the case, it returns the database,
+otherwise it creates it.
+*/
+Database initDatabase(string databasePath)
+{
+    import std.file : exists;
+    import ae.sys.file : ensurePathExists;
+
+    if (databasePath.exists)
+        return Database(databasePath);
+
+    ensurePathExists(databasePath);
+    return Database(databasePath, [table_schema,]);
+}
+
+void updateScoreBoard(string author, uint points, Database db = _db)
```

Comment:
Other tests will modify the "real" global database (due to calling `updateScoreBoard`) as part of their execution. This shouldn't be a problem in practice, because we don't run the tests in the same environment as running the "production" instance, but is a hypothetical problem otherwise. For this reason in my initial proposal I replaced the global `Database` object. Alternatively, the default database path could be set to something else in `version(unittest)` mode (which is already used to control whether e.g. to compile the `main` function).

Reply:
My reasoning was that unittests should provide the testing database, whereas "production" calls will use the global defined database. Any other solution would require to modify the `updateScoreBoard` function to special case the testing database. I can add a comment to highlight that.

Reply:
Then you would need to propagate the database object across the call chain starting from every `unittest` block and until `updateScoreBoard` and add it to the argument list of every function involved